tests/components/scrape/test_sensor.py

```diff
@@ -155,12 +159,13 @@ async def test_scrape_sensor_no_data_refresh(hass: HomeAssistant) -> None:
     assert state
     assert state.state == "Current Version: 2021.12.10"
 
-    mocker.data = None
-    await async_update_entity(hass, "sensor.ha_version")
+    mocker.payload = "test_scrape_sensor_no_data"
+    async_fire_time_changed(hass, datetime.utcnow() + SCAN_INTERVAL)
+    await hass.async_block_till_done()
 
-    assert mocker.data is None
+    state = hass.states.get("sensor.ha_version")
     assert state is not None
-    assert state.state == "Current Version: 2021.12.10"
+    assert state.state == STATE_UNAVAILABLE
```

Comment:
Note: there is a slight change of behavior here.
With the DataCoordinator, we now set the state as UNAVAILABLE, where previously it returned the last value.